### ISSUE TITLE
Main example doesn't work

### ISSUE THREAD

**Issue by rhyous** [Thwaitesy/MSTestHacks#11]
You main example doesn't work for me. I have VS 2013 Update 4 with Resharper and various other plugins. 

1. I created a new C# Unit Test project called MsTestHacksExample.
2. Installed MsTestHacks from NuGet.
3. Made sure app.config exists. (It was created by your NuGet package. Well, actually Newonsoft.Json created it).
4. Added this test.

```
using System;
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MSTestHacks;

namespace MsTestHacksExample
{
    [TestClass]
    public class UnitTest1 : TestBase
    {
        private IEnumerable<int> Stuff
        {
            get
            {
                //This could do anything, fetch a dynamic list from anywhere....
                return new List<int> { 1, 2, 3 };
            }
        }

        [TestMethod]
        [DataSource("MsTestHacksExample.UnitTest1.Stuff")]
        public void TestMethod1()
        {
            var number = this.TestContext.GetRuntimeDataSourceObject<int>();

            Assert.IsNotNull(number);
        }
    }
}
```

5. Pressed Ctrl + R, Ctrl + T and the Unit Test ran, but this was the result:

```
Test Name:	TestMethod1
Test FullName:	MsTestHacksExample.UnitTest1.TestMethod1
Test Source:	c:\Users\jbarneck\Documents\Visual Studio 2013\Projects\tmp\MsTestHacksExample\MsTestHacksExample\UnitTest1.cs : line 23
Test Outcome:	Failed
Test Duration:	0:00:00

Result Message:	The unit test adapter failed to connect to the data source or to read the data. For more information on troubleshooting this error, see "Troubleshooting Data-Driven Unit Tests" (http://go.microsoft.com/fwlink/?LinkId=62412) in the MSDN Library. Error details: Object reference not set to an instance of an object.
```

I am going to keep troubleshooting. Hopefully there is a simple step I am missing or something I've installed is interfering.

**Comment by Thwaitesy** [comment]
Odd, have you tried running VS as administrator?

Also have you pulled down the code and tried to run the test project and see if that works?

**Comment by rhyous** [comment]
Running as admin didn't change anything.

Pulling source now.

**Comment by rhyous** [comment]
Download your project. Compiled. Referenced compiled dll. Same issue.
Added my project to your project. Removed nuget package. Referenced your project, and it worked.

I created a second project. Added the nuget package, tried again. Same failure.
I added the compiled dll from latests source. Same issue.
Added second project to your solution.  Removed nuget package. Referenced your project, and it worked.

So something is wrong with your NuGet Package.

I found this log:

[08-05-15 04:34:41] ------------------
[08-05-15 04:34:41] Starting Execution
[08-05-15 04:34:42] MsTestHacksExample2.UnitTest1.Stuff: System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
   at MSTestHacks.RuntimeDataSource.AttachRuntimeDataSources.<>c__DisplayClass18.<.cctor>b__c(Object iteration)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Xml.Linq.XContainer.AddContentSkipNotify(Object content)
   at System.Xml.Linq.XContainer.Add(Object content)
   at MSTestHacks.RuntimeDataSource.AttachRuntimeDataSources..cctor() in c:\Users\jbarneck\Documents\Projects\GIT\MSTestHacks\MSTestHacks\RuntimeDataSource\AttachRuntimeDataSources.cs:line 120

=== Pre-bind state information ===
LOG: DisplayName = Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed
 (Fully-specified)
LOG: Appbase = file:///C:/Users/jbarneck/Documents/Visual Studio 2013/Projects/tmp/MsTestHacksExample2/MsTestHacksExample2/bin/Debug
LOG: Initial PrivatePath = NULL
Calling assembly : MSTestHacks, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null.
===
LOG: This bind starts in default load context.
LOG: Using application configuration file: C:\Users\jbarneck\Documents\Visual Studio 2013\Projects\tmp\MsTestHacksExample2\MsTestHacksExample2\bin\Debug\MsTestHacksExample2.dll.config
LOG: Using host configuration file: 
LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework\v4.0.30319\config\machine.config.
LOG: Post-policy reference: Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed
LOG: Attempting download of new URL file:///C:/Users/jbarneck/Documents/Visual Studio 2013/Projects/tmp/MsTestHacksExample2/MsTestHacksExample2/bin/Debug/Newtonsoft.Json.DLL.
WRN: Comparing the assembly name resulted in the mismatch: Major Version
ERR: Failed to complete setup of assembly (hr = 0x80131040). Probing terminated.

[08-05-15 04:34:42] No of Assemblies Searched: 14, No of DataSources Found: 1, No of Iterations Created: 0, Total Time Taken: 00:00:00.0834980.
[08-05-15 04:34:42] Finishing Execution
[08-05-15 04:34:42] ------------------

**Comment by rhyous** [comment]
Looks like your NuGet package is loading Newtosoft.Json version 5. But your dll is specifically looking for version 6.

I updated the Newtosoft.Json NuGet package and it now works.

**Comment by stamoun** [comment]
Updating the nuspec to use 6.0.8 should be fine (I'm not set up to create a pull request), this was most likely introduced in d8c4cb7...

**Comment by timabell** [comment]
Just tripped over this myself.

**Comment by timabell** [comment]
I think this is fixed in PR #10

**Comment by Thwaitesy** [comment]
This should now be fixed.
